File: src/main/java/com/ptah/common/impl/OSSService.java
package com.ptah.common.impl;

import cn.hutool.core.util.SerializeUtil;
import com.aliyun.oss.OSS;
import com.aliyun.oss.OSSClientBuilder;
import com.ptah.common.Errors;
import com.ptah.common.ObjectStorageService;
import com.ptah.common.exceptions.ApplicationException;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.io.InputStream;
import java.util.Optional;

@Service
public class OSSService implements ObjectStorageService {
    @Value("${oss.endpoint}")
    private String ossEndpoint;
    @Value("${oss.accessKeyId}")
    private String accessKeyId;
    @Value("${oss.accessKeySecret}")
    private String accessKeySecret;

    @Value("${oss.bucketName}")
    private String bucketName;

    @Override
    public <T> T getByKey(String key, Class<T> type) {
        OSS ossClient = null;
        try {
            ossClient = getOssClient();
            return SerializeUtil.deserialize(ossClient.getObject(bucketName, key).getObjectContent().readAllBytes(), type);
        } catch (Exception e) {
            throw new ApplicationException(Errors.INVALID_CREDENTIALS);
        } finally {
            Optional.ofNullable(ossClient).ifPresent(OSS::shutdown);
        }
    }

    private OSS getOssClient() {
        return new OSSClientBuilder().build(ossEndpoint, accessKeyId, accessKeySecret);
    }

    @Override
    public void setValue(String key, Object value) {

    }

    @Override
    public void setValue(String key, Object value, long timeout) {

    }

    @Override
    public boolean hasKey(String key) {
        return false;
    }
}

File: backend/src/main/kotlin/com/gihub/jarryzhou/ptah/common/impl/OSSService.kt
package com.gihub.jarryzhou.ptah.common.impl

import cn.hutool.core.util.SerializeUtil
import com.aliyun.oss.OSS
import com.aliyun.oss.OSSClientBuilder
import com.ptah.common.Errors
import com.ptah.common.ObjectStorageService
import com.ptah.common.exceptions.ApplicationException
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.util.*

@Service
class OSSService : ObjectStorageService {
    @Value("\${oss.endpoint:}")
    private val ossEndpoint: String? = null

    @Value("\${oss.accessKeyId:}")
    private val accessKeyId: String? = null

    @Value("\${oss.accessKeySecret:}")
    private val accessKeySecret: String? = null

    @Value("\${oss.bucketName:}")
    private val bucketName: String? = null
    override fun <T> getByKey(key: String?, type: Class<T>?): T {
        var ossClient: OSS? = null
        return try {
            ossClient = this.ossClient
            SerializeUtil.deserialize(ossClient.getObject(bucketName, key).objectContent.readAllBytes(), type)
        } catch (e: Exception) {
            throw ApplicationException(Errors.INVALID_CREDENTIALS)
        } finally {
            Optional.ofNullable(ossClient).ifPresent { obj: OSS -> obj.shutdown() }
        }
    }

    private val ossClient: OSS
        get() = OSSClientBuilder().build(ossEndpoint, accessKeyId, accessKeySecret)

    override fun setValue(key: String?, value: Any?) {}
    override fun setValue(key: String?, value: Any?, timeout: Long) {}
    override fun hasKey(key: String?): Boolean {
        return false
    }
}
